core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -874,6 +874,11 @@ public int performBatchRecovery(Iterable<Translog.Operation> operations) {
      * After the store has been recovered, we need to start the engine in order to apply operations
      */
     public void performTranslogRecovery(boolean indexExists) {
+        if (indexExists == false) {
+            // note: these are set when recovering from the translog
+            recoveryState().getTranslog().totalOperations(0);
```

Comment:
I looked at it and tried but we only do it in 2 places and make the translog recovery state stats a thing will introduce unneeded bloot imh. I simplified things a (tiny) bit.